src/Umbraco.Web/Hosting/AspNetHostingEnvironment.cs

```diff
@@ -44,6 +47,23 @@ public void LazyRestartApplication()
             HttpRuntime.UnloadAppDomain();
         }
 
+        private IDictionary<IRegisteredObject, RegisteredObjectWrapper> RegisteredObjects { get;  } = new Dictionary<IRegisteredObject, RegisteredObjectWrapper>();
```

Comment:
Why isn't this just a private readonly field and should probably be ConcurrentDictionary since we want to make sure this is 100% threadsafe.